Source/UnitEx.cpp

```diff
@@ -1129,14 +1138,19 @@ void UnitEx::SetAnimation(int animIdx, int animDelay, int animStartDelay, int bo
 	func(&(*unitArray)[unitID], 0, animIdx, animDelay, animStartDelay, boolInvisible, boolSkipDoDeath);
 }
 
+// HFL initialization must be checked before calling GetBeaconData
+BeaconData& GetBeaconData(int unitID)
+{
+	return *(BeaconData*)(reinterpret_cast<std::uintptr_t>(*unitArray) + (unitID * sizeof(OP2Unit)) + beaconDataOffset);
+}
+
 int UnitEx::GetNumTruckLoadsSoFar()

```

Comment:
Should all of the functions checking beacon data and lab data also be checking if the unit IsLive or not? I thought about adding these checks. The other unit functions in this class seem to check that regularly.

Reply:
That might be a discussion that ties to exception handling. I'd be fine leaving it out for now. Conceivably someone might want to check up on details of a unit that was recently destroyed, so there may be some benefit to not getting in the way.